shell/platform/linux/fl_engine.cc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_engine.h"
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+/**
+ * FlEngine:
+ *
+ * #FlEngine is an abstract class that controls a FlutterEngine.
+ */
```

Comment:
I'd strongly prefer class-level comments to go in the header unless this is a very strong convention.

While I'm thinking about it: is there a document somewhere that explains why function declaration comments need to go in the implementation files? Given that we'll be shipping the headers to clients, but not the implementation files, it seems really unfortunate not to have any documentation that's in the source. Yes, we'll be generating automatic docs, but that forces people to leave their IDE and find a web page, vs using jump-to-declaration. Surely generation could be done using comments on declarations instead?

Reply:
Yes, we agreed to do that before, but I'd forgotten. I'll move them all to the headers.

Reply:
I'd forgotten whether we'd actually discussed this, so we're even :)